File: frontend/src/api/configFiscal.ts
import api from './client'

export interface TemplateDescricao { tipo: string; label: string; texto: string }
export interface CodigoFavorito { codigo: string; label: string; descricao?: string }

export interface ConfigFiscal {
  // Emitente
  razao_social: string
  cnpj: string
  inscricao_municipal?: string
  cnae?: string
  endereco?: string
  municipio_ibge: string
  municipio_nome: string
  uf: string
  email_fiscal?: string
  telefone_fiscal?: string
  regime_tributario: string
  // Bases
  aliquota_iss: number
  regime_especial: string
  anexo_simples: string
  rbt12?: number
  aliquota_efetiva_simples: number
  ret_ir_pct: number
  ret_inss_pct: number
  ret_csll_pct: number
  ret_pis_pct: number
  ret_cofins_pct: number
  iss_retido_padrao: boolean
  // IBS/CBS
  ibs_pct: number
  cbs_pct: number
  piloto_ibscbs: boolean
  // Contabilidade
  emails_contador: string[]
  email_master?: string
  dia_envio_relatorio: number
  enviar_relatorio_auto: boolean
  // Numeração
  serie_padrao: string
  // Catálogos
  codigos_favoritos: CodigoFavorito[]
  templates_descricao: TemplateDescricao[]
  // DANFSe
  rodape_danfse?: string
  // Calculados (read-only)
  aliquota_simples_sugerida?: number
  faixa_simples?: string
  updated_at?: string
}

export const configFiscalApi = {
  obter: () => api.get<ConfigFiscal>('/config-fiscal').then((r) => r.data),
  salvar: (data: ConfigFiscal) => api.put<ConfigFiscal>('/config-fiscal', data).then((r) => r.data),
}

File: backend/app/schemas/config_fiscal.py
from __future__ import annotations

from datetime import datetime
from decimal import Decimal
from typing import Optional

from pydantic import BaseModel


class TemplateDescricao(BaseModel):
    tipo: str
    label: str
    texto: str


class CodigoFavorito(BaseModel):
    codigo: str
    label: str
    descricao: str = ""


class ConfigFiscalBase(BaseModel):
    # Emitente
    razao_social: str
    cnpj: str
    inscricao_municipal: Optional[str] = None
    cnae: Optional[str] = None
    endereco: Optional[str] = None
    municipio_ibge: str
    municipio_nome: str
    uf: str
    email_fiscal: Optional[str] = None
    telefone_fiscal: Optional[str] = None
    regime_tributario: str

    # Bases
    aliquota_iss: Decimal
    regime_especial: str
    anexo_simples: str
    rbt12: Optional[Decimal] = None
    aliquota_efetiva_simples: Decimal
    ret_ir_pct: Decimal
    ret_inss_pct: Decimal
    ret_csll_pct: Decimal
    ret_pis_pct: Decimal
    ret_cofins_pct: Decimal
    iss_retido_padrao: bool

    # IBS/CBS
    ibs_pct: Decimal
    cbs_pct: Decimal
    piloto_ibscbs: bool

    # Contabilidade
    emails_contador: list[str] = []
    email_master: Optional[str] = None
    dia_envio_relatorio: int
    enviar_relatorio_auto: bool

    # Numeração
    serie_padrao: str

    # Catálogos
    codigos_favoritos: list[CodigoFavorito] = []
    templates_descricao: list[TemplateDescricao] = []

    # DANFSe
    rodape_danfse: Optional[str] = None
    danfse_prefixo_numero: str = ""
    carga_media_pct: Decimal = Decimal("16.33")


class ConfigFiscalOut(ConfigFiscalBase):
    updated_at: Optional[datetime] = None
    # Sugestão calculada (não persiste) — preenchida pelo router
    aliquota_simples_sugerida: Optional[Decimal] = None
    faixa_simples: Optional[str] = None
    rbt12_acumulado_12m: Optional[float] = None
    aliquota_pelo_acumulado: Optional[float] = None
    link_publico_token: Optional[str] = None

    model_config = {"from_attributes": True}


class ConfigFiscalUpdate(ConfigFiscalBase):
    pass
